File: datastructures lab/practice/STACK_LL.c
/******************************************************************************
    C Program To Implement STACK Using Linked List.
*******************************************************************************/

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node *link;
};
struct node *first = NULL;

void push()
{
    int item;
    struct node *p;
    p = malloc(sizeof(struct node));
    printf("Enter The Element To be Pushed : ");
    scanf("%d",&item);
    p->data = item;
    p->link = NULL;
    
    if(first == NULL)
        first = p;
    else
    {
        p->link = first;
        first = p;
    }
}

void pop()
{
    struct node *temp;
    if(first == NULL)
        printf("POP Operation Not Possibl.!");
    else if(first->link == NULL)
    {
        printf("Element Deleted is %d",first->data);   
        free(first);
        first = NULL;
    }
    else
    {
        temp = first;
        first = first->link;
        printf("Element Deleted is %d",temp->data);
        free(temp);
    }
}

void display()
{
    struct node *temp;
    if(first == NULL)
        printf("Display Operation Not Possible.!");
    else
    {
        temp = first;
        while(temp != NULL)
        {
            printf("\t\n%d",temp->data);
            temp = temp->link;
        }
    }
}

int main()
{
    int choice;
    printf("STACK Implementation Using Linked List : ");
    while(1)
    {
        printf("\nThe Choices Are : \n1. PUSH.\n2. POP.\n3. Display.\n4. Exit.\nEnter Your Choice : ");
        scanf("%d",&choice);
        
        switch(choice)
        {
            case 1 : push();
                     break;
                     
            case 2 : pop();
                     break;
                     
            case 3 : display();
                     break;
                     
            case 4 : exit(0);
                     break;
                     
            default : printf("Invalid choice.!");         
        }
    }
    return 0;
}

File: OOJava_lab/DataStructures/STACK_LL.java
/******************************************************************************    
    Java Program To implement STACK Using Linked List.        
*******************************************************************************/

import java.util.Scanner;
import java.util.LinkedList;

class Stack
{
    LinkedList<Integer> stck = new LinkedList<Integer>();
    
    public void push()
    {
        System.out.println("Enter The Item To Be Inserted : ");
        Scanner sc = new Scanner(System.in);
        int item = sc.nextInt();
        stck.addFirst(item);
    }
    
    public void pop()
    {
        if(stck.size() == 0)
        {
            System.out.println("POP Operation Not Possible, STACK UNDERFLOW.");
        }
        else
        {
            System.out.println("Element Deleted is "+stck.removeFirst());
        }
    }
    
    public void display()
    {
        if(stck.size() == 0)
        {
            System.out.println("Display Operation Not Possible, STACK is EMPTY.!");
        }
        else
        {
            System.out.println(stck);   
        }
    }
}

public class STACK_LL
{
	public static void main(String[] args) 
	{
		Stack s = new Stack();
		
		System.out.println("STACK Impelmentation Using LinkedList : ");
		
		while(true)
		{
		    System.out.println("The Chocies Are : ");
		    System.out.println("1. PUSH. \n2. POP. \n3. Display. \n4. Exit");    
		    System.out.println("\nEnter Your Choice : ");
		    Scanner sc = new Scanner(System.in);
		    int choice = sc.nextInt();
		    
		    switch(choice)
		    {
		        case 1 : s.push();
		                 break;
		               
		        case 2 : s.pop();
		                 break;
		                 
		        case 3 : s.display();
		                 break;
		                 
		        case 4 : System.exit(0);
		                 break;
		        
		        default: System.out.println("Invalid Choice..!");
		    }
		}
	}
}
